A standalone tutorial for `TB_to_sympy`.

Collins-kariuk opened the pull request:
I would like to contribute to the TRIQS tutorial repository, an accompanying tutorial I wrote with the help of my mentors Olivier Gingras and Sophie Beck regarding my `TB_to_sympy` which returns the analytical form of the momentum space Hamiltonian of the tight-binding model from a tight-binding lattice object by utilizing the Fourier series.

This function has already been merged into TRIQS and can be found in the utils.py file. The tutorial provides the user with a brief introduction to the tight-binding model and enables the user to interact with different parameters of my function; as well as comparing minimal tight-binding models.

Collins-kariuk committed:
Included the sympyfy function alongside its dependencies

Collins-kariuk committed:
Create .DS_store

Collins-kariuk committed:
Deleted the notebook containing the sympyfy function

Collins-kariuk committed:
Started working on the tutorial

Collins-kariuk committed:
Added DS Store (folder metadata) to gitignore

Collins-kariuk committed:
Wrote the heading and first text cell

Collins-kariuk committed:
Wrote about sympy's parameters and TB Hamiltonian

Collins-kariuk committed:
Finalized first draft

Collins-kariuk committed:
Finalized first draft of tutorial. Added example output and conclusion.

Collins-kariuk committed:
Amended the statement that the TB model is only suitable for weakly-correlated systems.

Collins-kariuk committed:
deleted the HTML import

Collins-kariuk committed:
Cut out the section on calling the function on the Strontium compound.

Collins-kariuk committed:
Added the _centres.xyz file for La2CuO4

Collins-kariuk committed:
Heavily changed the tutorial according to Olivier's recs. Check the description:

- Introduced an example where I call sympyfy on a simple-structured Lathanum Cuprate material under different scenarios.
- Changed the introduction to include the fact that w90 includes all hopping parameters but that the user has the option of using PythTB to include cutoffs on the hopping amplitude and hopping distance.
- Made sure to highlight the fact that the numerical expression isn't wholly numerical but depends on the k-space vector variables.

Collins-kariuk committed:
Added the reason why the Hamiltonian is 1-by-1

Collins-kariuk committed:
Made some changes to the introduction and introduced a band structure of the full vs cutoff model of the Lanthanum cuprate compound

Collins-kariuk committed:
Added La2CuO4 band structure comparing full W90 model vs cutoff model

Collins-kariuk committed:
Deleted this because it is based off wrong parameters

Collins-kariuk committed:
Showed how I call the model and included additional band structure figure (at 0.01 eV min_hopping_norm)

Collins-kariuk committed:
Changed the form of the Hamiltonian equation and implemented reductions to the text in the introduction

Collins-kariuk committed:
Created a python file containing the TB_to_sympy function

So that I could import it into my tutorial.

Collins-kariuk committed:
included the necessary dependencies

Collins-kariuk committed:
Made a figure showing band structure of lanthanum cuprate at 2 different energy cutoffs

Collins-kariuk committed:
Cut down on the text and merged the 2 band structure images

Collins-kariuk committed:
chopping down unnecessary comments and words in docstring

Collins-kariuk committed:
changed the location of where I import sympy to inside the function

Collins-kariuk committed:
changed the order for one of the sentences to promote better flow in the tutorial

Collins-kariuk committed:
Succinctly phrased the title and some parts of the introduction to clearly convey the message

Collins-kariuk committed:
changed the name of the first parameter and changed some variable names accordingly

Collins-kariuk committed:
Added figure that compares minimal models of La2CuO4

Collins-kariuk committed:
Added notes to intro (from poster) and replaced outdated minimal model figure

Collins-kariuk committed:
Finished first draft of the section on comparing minimal models

Collins-kariuk committed:
No longer needed

Collins-kariuk committed:
Rewrote H(k)

Collins-kariuk committed:
No longer needed

Collins-kariuk committed:
Created a Python file from which we can call our main function from the tutorial notebook

Collins-kariuk committed:
Improved how we get the plot by getting rid of the inefficient function by replacing it with a for loop that basically does the same plotting work for us

Collins-kariuk committed:
Go rid of remaining \bolds

Collins-kariuk committed:
Updating the calling of TB_to_sympy

Collins-kariuk committed:
Update Hamiltonian expression & importations

the-hampel committed:
remove TB_to_sympy.py file and use triqs function

the-hampel committed:
update tutorial with triqs module

the-hampel committed:
remove gitignore changes

the-hampel committed:
remove unneccesary files

the-hampel committed:
corrections for tutorial

Collins-kariuk commented:
Hello @the-hampel. Thanks for approving my changes. Shall I close this PR? What is remaining?

the-hampel commented:
Hi @Collins-kariuk, @Wentzell will have a final look and than we will merge the tutorial. So please leave it open for now. Thanks!

Wentzell commented on Advanced/TB_to_sympy.ipynb:
````diff
@@ -0,0 +1,416 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The ```TB_to_sympy``` function takes hopping amplitudes to generate a TB model. The hopping amplitudes are obtained from the Wannier90 package, which fits the band structure obtained from density functional theory (DFT) for a set of orbitals. The larger the hopping amplitude, the more significant it is. For instance, a hopping amplitude of 0.00005 eV does not contribute as much to the moveme ...
````
This is not quite correct? It does not generate a TB model, it takes a pre-constructed one. Also it could be obtained out of a Wannier90 calculation or could just be a model Hamiltonian.
The output is an analytic expression of that same TB model.

Collins-kariuk replied:
Fixed.

Wentzell commented on Advanced/TB_to_sympy.ipynb:
```diff
@@ -0,0 +1,416 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In second quantization, the TB Hamiltonian is given by:\n",
```
You mean 'a generic form of a TB Hamiltonian reads' ?

Collins-kariuk replied:
Fixed.

Wentzell commented on Advanced/TB_to_sympy.ipynb:
````diff
@@ -0,0 +1,416 @@
+    "- $\\sigma$ is the spin and $c^{\\dagger}$\n",
+    "- $c$ are creation and annihilation operators.\n",
+    "\n",
+    "In this tutorial, we examine a system with one orbital and one site per unit cell. By performing a Fourier transform of the Hamiltonian from real space to momentum space, we obtain $H(k)$, which is achieved using the ```TB_to_sympy``` function.\n",
````
I would drop here the part ", which is achieved using the ```TB_to_sympy``` function."

Collins-kariuk replied:
Dropped.

Wentzell commented on Advanced/TB_to_sympy.ipynb:
````diff
@@ -0,0 +1,416 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```TB_to_sympy``` is defined as:"
````
Change to: 'The documentation of the ```TB_to_sympy``` function reads'

Also, why don't you use `?TB_to_sympy` to generate documentation?

Collins-kariuk replied:
Changed.

Collins-kariuk replied:
I have also now used ?TB_to_sympy to generate documentation. Thank you for this. I did not know one could do this.

Wentzell commented on Advanced/TB_to_sympy.ipynb:
````diff
@@ -0,0 +1,416 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The only required parameter for our function is ```TBL```, a TRIQS TBLattice Object (Tight-Binding Lattice Object). To obtain ```TBL```, you can use TRIQS-provided functions that allow conversion of your model to a TRIQS TBLattice object. These functions are:\n",
````
The TBL doesn't need to come out of a conversion, it can be directly defined.

Collins-kariuk replied:
Fixed.

Wentzell commented on Advanced/TB_to_sympy.ipynb:
````diff
@@ -0,0 +1,416 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The main distinction between the analytical and numerical expressions is that the numerical expression depends on ```kx```, ```ky```, and ```kz```, while the analytical expression additionally depends on the lattice vectors, ```a1```, ```a2```, and ```a3```. The lattice vectors define the orientation and shape of the crystal unit cell and can be replaced with numerical values using the ```un ...
````
This is not clear. You should state that the numerical expression is uses euclidian unit vectors and the analytical expression uses reciprocal basis vectors `ai`

Collins-kariuk replied:
Fixed.

Wentzell commented on Advanced/TB_to_sympy.ipynb:
```diff
@@ -0,0 +1,416 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def minimal_model(w90_input, fermi_ev, cutoff):\n",
```
What kind of object is `w90_input` here?

Collins-kariuk replied:
Added a docstring to the `minimal_model` function to make it clear.

Collins-kariuk committed:
Implemented Wentzell's proposed changes.

Collins-kariuk committed:
Added the tutorial after implementing changes

This was done after the whole rebasing fiasco. Hopefully, it didn't alter anything major. I had to delete the tutorial notebook first as I was rebasing which is why I am adding the tutorial wholly now.

Collins-kariuk commented:
Hello @Wentzell , I have implemented your proposed changes. They are visible in the Sympyfy analytic expresser branch. I am not sure whether I should open another pull request so I have not. If you need me to, I will. Let me know if there are other things I can do.
